### DISCUSSION TITLE
Self-hosting not working properly

### DISCUSSION THREAD

**Discussion by nokia6290** [supabase/supabase#20111] [General]
Hi all,

Trying to self-host supabase.

Tutorials, videos, official documentation is really challenging and sometimes nonexistant ...

However. Got it eventually running ( supabase, edited production env files with secrets and all. Launched successfully studio login on digital ocean Ec2 instance. (2cpu, 4gb ram, enough ssd)

Built a mini website on flutter and trying to fetch some basic data (no auth, and no security on them added so the API call is publicly available.).

Built a simple table on public scheme..made a query that fetches all my data that I want.

My final log on client side is:
state is :: ErrorState(Error fetching all tools :: Exception: Error fetching exercises :: ClientException: XMLHttpRequest error., uri=http://170.64.202.143:8000/rest/v1/rpc/get_ai_tool_data)

Logging docker compose: ( you can postgress is healthy, but studio is unhealthy and I'm not sure why.

root@ubuntu-s-2vcpu-4gb-sfo3-01:~/supabase/docker# docker compose ps
NAME                             COMMAND                   SERVICE             STATUS                PORTS
realtime-dev.supabase-realtime   "/usr/bin/tini -s -g…"    realtime            running (healthy)     
supabase-analytics               "sh run.sh"               analytics           running (healthy)     0.0.0.0:4000->4000/tcp, :::4000->4000/tcp
supabase-auth                    "gotrue"                  auth                running (healthy)     
supabase-db                      "docker-entrypoint.s…"    db                  running (healthy)     0.0.0.0:5432->5432/tcp, :::5432->5432/tcp
supabase-edge-functions          "edge-runtime start …"    functions           running               
supabase-imgproxy                "imgproxy"                imgproxy            running (healthy)     8080/tcp
supabase-kong                    "bash -c 'eval \"echo…"   kong                running (healthy)     0.0.0.0:8000->8000/tcp, 0.0.0.0:8443->8443/tcp, :::8000->8000/tcp, :::8443->8443/tcp
supabase-meta                    "docker-entrypoint.s…"    meta                running (healthy)     8080/tcp
supabase-rest                    "postgrest"               rest                running               3000/tcp
supabase-storage                 "docker-entrypoint.s…"    storage             running (healthy)     5000/tcp
supabase-studio                  "docker-entrypoint.s…"    studio              running (unhealthy)   3000/tcp
supabase-vector                  "/usr/local/bin/vect…"    vector              running (healthy)     

another log:
PostgreSQL Database directory appears to contain a database; Skipping initialization

I'm googling for hours and can't understand what is wrong with this supabase self-host. Makes me think its overrated with marketing tutorials but in reality its full of bugs and missing tutorials.

The creators probably are clever enough to add many obstalces in order to use their main paid supabase service..

Sorry for frustration.

Anyone else had issues, or particularly similar issue?

Are we supposed to configure other files in the supabase that no one tells you to do that?

**Comment by saltcod** [comment]
Hi @nokia6290 — moved to a discussion and renamed to be more descriptive. 

What guide did you follow?

**Comment by TimchaStudio** [comment]
Has your issue been resolved? I have encountered the same error as well.

"I haven't modified any code. Simply following the [official Supabase Docker tutorial](https://supabase.com/docs/guides/self-hosting/docker) for local installation results in the same error."

ERROR

```
172.21.0.3 - - [12/Mar/2024:13:00:47 +0000] "POST /rest/v1/rpc/match_file_items_openai HTTP/1.1" 404 226 "-" "node"
```

**Comment by sanbei101** [comment]
**是的,我也遇到了同样的问题,在获取日志时也遇到了问题,要知道这是一台刚刚初始化的云服务器,刚刚装上supabase**
```
{
  "code": 502,
  "errors": [],
  "message": "Something went wrong! Unknown error. If this continues please contact support.",
  "status": "UNKNOWN"
}
```

**Reply by viv1** [reply]
Facing the same issue as listed above . Followed [Self hosting using Docker guide](https://supabase.com/docs/guides/self-hosting/docker)

**Comment by K-NRS** [comment]
Same case we're facing too

**Comment by SillyHippy** [comment]
same issue
